### PR TITLE
chore: release 1.0 proposal

### PR DESCRIPTION
**Pull request by dyladan**
### :memo: Documentation

* [#89](https://github.com/open-telemetry/opentelemetry-js-api/pull/89) chore: update upgrade guidelines ([@dyladan](https://github.com/dyladan))

### :house: Internal

* [#90](https://github.com/open-telemetry/opentelemetry-js-api/pull/90) chore: enable typescript 4.3 noImplicitOverride option ([@Flarna](https://github.com/Flarna))

### Committers: 2

* Daniel Dyla ([@dyladan](https://github.com/dyladan))
* Gerhard Stöbich ([@Flarna](https://github.com/Flarna))

### CODE REVIEW

**Review by MSNev — APPROVED** [review]

**Review by aabmass — APPROVED** [review]
Super exciting, thanks for all the hard work everyone! 🚀

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@opentelemetry/api",
-  "version": "0.21.0",
+  "version": "0.22.0",
```
Comment (Flarna) [inline package.json:3]: I guess this should be 1.0.0.

File: README.md [hunk 2]
```diff
@@ -105,6 +105,10 @@ Because the npm installer and node module resolution algorithm could potentially
 
 ## Upgrade Guidelines
 
+### 0.21.0 to 1.0
```
Comment (Flarna) [inline README.md:108]:
```suggestion
### 0.21.0 to 1.0.0
```

File: package.json (hunk 1 above)
Comment (dyladan) [reply package.json:3]: heh thanks

**Review by Flarna — APPROVED** [review]

File: test/internal/global.test.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ const api2 = require('../../src') as typeof import('../../src');
 
 // This will need to be changed manually on major version changes.
 // It is intentionally not autogenerated to ensure the author of the change is aware of what they are doing.
-const GLOBAL_API_SYMBOL_KEY = 'io.opentelemetry.js.api.0';
+const GLOBAL_API_SYMBOL_KEY = 'io.opentelemetry.js.api.1';
```
Comment (Flarna) [inline test/internal/global.test.ts:34]: I think this means we pretend compatibility with the outdated 1.0.0-rc.X releases which is not true.

Comment (dyladan) [reply test/internal/global.test.ts:34]: I'll change the name for now. We can release a minor change later which enforces strict equality for prerelease versions which will give us the ability to add canary releases in the future if we want them

Comment (dyladan) [reply test/internal/global.test.ts]: https://github.com/open-telemetry/opentelemetry-js-api/pull/94

**Review by obecny — APPROVED** [review]
lgtm, yay!

**Review by vmarchaud — APPROVED** [review]

**Review by weyert — APPROVED** [review]

**Pull request merged**